lib/intf.js

```diff
@@ -267,26 +290,65 @@ export default function (self, ctor) {
     }
     return f;
   };
-  self.applyFilter = function () {
-    self.refreshFromOrigialData();
-    Object.keys(self.columnFilters).forEach(function (filter) {
-      var header = self.getHeaderByName(filter);
+  self.getFilteredAndSortedViewData = function (originalData) {
+    // We make a copy of originalData here in order be able to
+    // filter and sort rows without modifying the original array.
+    // Each row is turned into a (row, rowIndex) tuple
+    // so that when we apply filters, we can refer back to the
+    // row's original row number in originalData. This becomes
+    // useful when emitting cell events.
+    let newViewData = originalData.map((row, originalRowIndex) => [
```

Comment:
What happens when data contains 10MM rows?  Are we making a copy of 100% of that data?  Is there a way to do this without copying the data?  Making a difference somehow?  I think this will cut performance or at least memory usage in half for larger record sets.  Shouldn't be hard to see what sort of overhead we're adding here.

Reply:
It shouldn't be copying the data: the `map` creates a new array, but `row` in `[row, originalRowIndex]` refers to the same object (i.e. deep-equals) as in the `originalData` array.

There _is_ a slight increase in memory usage, as we're keeping track of `originalRowIndex`, and this is then stored in the `boundRowIndexMap`, but that's `O(2N)` (~8 bytes for the `viewRowIndex` as key, and ~8 bytes for the value), so I think that's acceptable? If I find time later, I'll see if I can set up a benchmark.